src/Plotting/plot_recipes.jl

```diff
@@ -313,7 +357,8 @@ julia> plot(Singleton([0.5, 1.0]))
         ylims --> lims[:y]
     end
 
-    plot_recipe(S, ε)
+    p = element(S)
+    [p[1], p[2]]
```

Comment:
1D plots are broken, and I think this call to `plot_recipe` was supposed to resolve that.
```julia
julia> B = (1 .. 2);
julia> Bs1 = LazySets.sample(LazySets.Interval(B), 1000);
julia> Plots.plot(Singleton.(Bs1))
ERROR: BoundsError: attempt to access 1-element Array{Float64,1} at index [2]
```

Reply:
good catch; i had in mind 2d plots, but we can special case that.

Reply:
This method does exactly that.

https://github.com/JuliaReach/LazySets.jl/blob/9d8a38773c568d7e2a29a21aff6d79794b5750f9/src/Interfaces/AbstractSingleton.jl#L369-L377

Reply:
do you think we can in some way reuse that function?

this PR was to improve the plotting recipe of vectors of a large number of singletons? the main difference of this PR with the one in master is that we make a scatter plot and we treat all the singletons in the same loop. the performance is very good (same as using Plots directly).

Reply:
I tried undoing this change and it gives the same result with exactly the same allocations, so I would say "yes" :smiley:
Was there a reason why you did not undo this change?

Reply:
ah, ok i see now! yes, i'll undo this change. the reason was that it was very unexpected to see `plot_recipe(S, ε)` where i just expected `[p[1], p[2]]` so i thought this could be introducing additional allocations etc.

Reply:
I agree that this is not very transparent :sweat_smile:

Reply:
i mean, the epsilon stuff seemed complicated for only returning the array.

Reply:
> and it gives the same result with exactly the same allocations

Well, that was not surprising because it does not trigger the discussed code anymore :roll_eyes: